File: testing_codes/C++/2-InputEmailToShell.cpp
#include <iostream>
#include <cstdlib>
#include <string>

int main() {
    std::string email;
    std::cout << "Enter your email address: ";
    std::getline(std::cin, email);

    if (email.find('@') == std::string::npos) {
        std::cout << "Invalid Email" << std::endl;
        return 1;
    }

    std::string command = "program.py " + email;
    system(command.c_str());

    return 0;
}

File: testing_codes/Java/2-InputEmailToShell.java
import java.util.Scanner;
import java.io.IOException;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter your email address: ");
        String email = scanner.nextLine();

        if (!email.contains("@")) {
            System.out.println("Invalid Email");
            System.exit(1);
        }

        try {
            ProcessBuilder processBuilder = new ProcessBuilder("program.py", email);
            processBuilder.start();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
